src/Components/Server/src/Circuits/CircuitHost.cs

```diff
@@ -324,7 +327,7 @@ public async Task OnRenderCompletedAsync(long renderId, string errorMessageOrNul
 
         try
         {
-            _ = Renderer.OnRenderCompletedAsync(renderId, errorMessageOrNull);
+            _ = HandleInboundActivityAsync(() => Renderer.OnRenderCompletedAsync(renderId, errorMessageOrNull));
```

Comment:
The following is nitpicky so feel free to ignore it, especially if you think it makes the code unpleasant!

It may be possible to avoid the extra allocation caused by capturing these locals in the closure by redefining `HandleInboundActivityAsync` as `HandleInboundActivityAsync<T>(T data, Func<T, Task> callback)`, then passing in the otherwise-captured values as the `T data` parameter (for example, the caller could pass a tuple containing whatever combination of values they want).

Not sure if that's justified or would make the code hard to read.

Reply:
I think in order to make this work, we would have to include `data` in the `CircuitInboundActivityContext` class so it can be passed all the way through to the last handler (otherwise a capture would have to happen somewhere else). But then we would probably need to make `CircuitInboundActivityContext` generic in order to avoid boxing the `data` parameter. And then we would need to introduce a non-generic base type for `CircuitInboundActivityContext` to avoid also making `IHandleCircuitActivity.HandleInboundActivityAsync` generic.

Or maybe there's a way simpler way of doing this that I'm missing, but if it really does require that we do all this ^ then I think it might be best to leave as is. Is that okay?

Reply:
I am ok if we leave it as is for now, and file an issue to revisit. If in the case that nobody registers an `IHandleCircuitActivity` implementation we don't allocate, I think it is ok to leave it like this for now.